frontend/pages/hosts/ManageHostsPage/ManageHostsPage.jsx

```diff
@@ -389,64 +479,47 @@ export class ManageHostsPage extends PureComponent {
     this.setState({ isAllMatchingHostsSelected: false });
   };
 
-  getNextLocationUrl = (
-    pathPrefix = "",
-    newRouteTemplate = "",
-    newRouteParams = {},
-    newQueryParams = {}
-  ) => {
-    const routeTemplate = newRouteTemplate || this.props.routeTemplate || "";
-    const urlRouteParams = Object.assign(
-      {},
-      this.props.routeParams,
-      newRouteParams
-    );
-    const urlQueryParams = Object.assign(
-      {},
-      this.props.queryParams,
-      newQueryParams
-    );
+  getLabelSelected = () => {
+    const { selectedFilters } = this.props;
+    return selectedFilters.find((f) => f.includes(LABEL_SLUG_PREFIX));
+  };
 
-    let routeString = "";
+  getStatusSelected = () => {
+    const { selectedFilters } = this.props;
+    return selectedFilters.find((f) => !f.includes(LABEL_SLUG_PREFIX));
+  };
 
-    if (!isEmpty(urlRouteParams)) {
-      routeString = reduce(
-        urlRouteParams,
-        (string, value, key) => {
-          return string.replace(`:${key}`, encodeURIComponent(value));
-        },
-        routeTemplate
-      );
-    }
+  getValidatedTeamId = (teamId) => {
+    const { currentUser, isOnGlobalTeam, teams } = this.props;
+
+    teamId = parseInt(teamId, 10);
 
-    let queryString = "";
-    if (!isEmpty(urlQueryParams)) {
-      queryString = reduce(
-        urlQueryParams,
-        (arr, value, key) => {
-          key && arr.push(`${key}=${encodeURIComponent(value)}`);
-          return arr;
-        },
-        []
-      ).join("&");
+    let currentUserTeams = [];
+    if (isOnGlobalTeam) {
+      currentUserTeams = teams;
+    } else if (currentUser && currentUser.teams) {
+      currentUserTeams = currentUser.teams;
     }
 
-    const nextLocation = union(
-      trim(pathPrefix, "/").split("/"),
-      routeString.split("/")
-    ).join("/");
+    const currentUserTeamIds = currentUserTeams.map((t) => t.id);
 
-    return queryString ? `/${nextLocation}?${queryString}` : `/${nextLocation}`;
-  };
+    const validatedTeamId =
+      !isNaN(teamId) && teamId > 0 && currentUserTeamIds.includes(teamId)
+        ? teamId
+        : 0;
 
-  getLabelSelected = () => {
-    const { selectedFilters } = this.props;
-    return selectedFilters.find((f) => f.includes(LABEL_SLUG_PREFIX));
+    return validatedTeamId;
   };
 
-  getStatusSelected = () => {
-    const { selectedFilters } = this.props;
-    return selectedFilters.find((f) => !f.includes(LABEL_SLUG_PREFIX));
+  setLastLocation = () => {
```

Comment:
Seeing this makes me wonder whether we can take advantage of the browser [History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API)?

Reply:
Wouldn't that result in a full page reload? My understanding is that a push to react-router alters the URL of the SPA but doesn't necessarily reload the entire page like a true back button action would.

Reply:
Ah, sorry I wasn't that clear. What I am actually imagining is that we use the History API via `react-router-redux`, so instead of doing `setLastLocation` and restoring the location from that, can we just do the `push` like we are already doing, and then `goBack` (https://github.com/reactjs/react-router-redux#pushlocation-replacelocation-gonumber-goback-goforward) when we want to restore?

If that works, it pushes a lot of the state management to the browser and simplifies things a good bit.